core/boot/src/main/java/org/eclipse/dataspaceconnector/boot/system/ExtensionLoader.java

```diff
@@ -92,8 +93,20 @@ public static void loadVault(ServiceExtensionContext context) {
     }
 
     public static @NotNull Telemetry loadTelemetry() {
-        // TODO: enable overriding telemetry through an extension
-        return new Telemetry(GlobalOpenTelemetry.get());
+        return new Telemetry(loadOpenTelemetry());
+    }
+
+    public static @NotNull OpenTelemetry loadOpenTelemetry() {
```

Comment:
1. This method doesn't need to be public, it's used only in this class.
2. nit: Can we inline it? it would reduce some code: 

```java
public static @NotNull Telemetry loadTelemetry() {
        var loader = ServiceLoader.load(OpenTelemetry.class);
        var openTelemetries = loader.stream().map(ServiceLoader.Provider::get).collect(Collectors.toList());
        return new Telemetry(loadOpenTelemetry(openTelemetries));
    }
 ```

Reply:
I can inline it yes thanks good idea!
As I remove the `loadOpenTelemetry` function, I will not make anything private as loadTelemetry needs to be public.